theano/tensor/signal/downsample.py

```diff
@@ -1,1088 +1,57 @@
-                                                  self.padding[1])
-                        col_end = builtins.min(col_st + ds1, img_cols)
-                        if sum_mode:
-                            val = gz[n, k, r, c]
-                        else:
-                            val = gz[n, k, r, c] / ((row_end - row_st) *
-                                                    (col_end - col_st))
-                        gx[n, k, row_st:row_end, col_st:col_end] += val
-        # unpad the image
-        gx = gx[:, :, pad_h:(img_rows - pad_h), pad_w:(img_cols - pad_w)]
-        gx_stg[0] = gx
 
-    def grad(self, inp, grads):
-        x, gz = inp
-        ggx, = grads
-        return [theano.tensor.zeros_like(x),
-                DownsampleFactorMax(
-                    self.ds, ignore_border=self.ignore_border,
-                    st=self.st, padding=self.padding, mode=self.mode)(ggx)]
+def max_pool_2d_same_size(*args, **kwargs):
+    warnings.warn("function 'max_pool_2d_same_size' is now located "
+                  "in 'pool'.")
+    return pool.max_pool_2d_same_size(*args, **kwargs)
 
 
-class DownsampleFactorMaxGradGrad(Op):
-    __props__ = ('ds', 'ignore_border', 'st', 'padding', 'mode')
+def max_pool_2d(*args, **kwargs):
+    warnings.warn("function 'max_pool_2d' has been renamed and is now located "
+                  "in 'pool.pool_2d'.")
+    return pool.pool_2d(*args, **kwargs)
 
-    def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max'):
-        self.ds = tuple(ds)
-        if not all([isinstance(d, int) for d in ds]):
-            raise ValueError(
-                "DownsampleFactorMax downsample parameters must be ints."
-                " Got %s" % str(ds))
-        if st is None:
-            st = ds
-        assert isinstance(st, (tuple, list))
-        self.st = tuple(st)
-        self.ignore_border = ignore_border
-        self.padding = tuple(padding)
-        if self.padding != (0, 0) and not ignore_border:
-            raise NotImplementedError(
-                'padding works only with ignore_border=True')
-        if self.padding[0] >= self.ds[0] or self.padding[1] >= self.ds[1]:
-            raise NotImplementedError(
-                'padding_h and padding_w must be smaller than strides')
-        self.mode = mode
-        assert self.mode == 'max'
 
-    def make_node(self, x, maxout, gz):
-        # make_node should only be called by the grad function of
-        # MaxPoolGrad, so these asserts should not fail.
-        x = tensor.as_tensor_variable(x)
-        maxout = tensor.as_tensor_variable(maxout)
-        gz = tensor.as_tensor_variable(gz)
-        assert x.ndim == 4
-        assert maxout.ndim == 4
-        assert gz.ndim == 4
+def DownsampleFactorMax(*args, **kwargs):
+    warnings.warn("Class 'DownsampleFactorMax' has been renamed and is now located "
+                  "in 'pool.Pool'.")
+    return pool.Pool(*args, **kwargs)
```

Comment:
I don't think it would, since the `__init__` method is not called when unpickling, but other methods of the object are called, which do not exist in a function.